### PR TITLE
feature/mx-1590 deterministic identity provider

### PR DESCRIPTION
**Pull request by rababerladuseladim**
# Changes
- make memory identity provider deterministic (same input args results in same stableTargetId and Identifier)

### CODE REVIEW

File: mex/common/identity/memory.py [hunk 1]
```diff
@@ -46,11 +48,23 @@ def assign(
         if identities:
             return identities[0]
 
+        def get_identifier(target_field: str) -> Identifier:
```
Comment (cutoffthetop) [inline mex/common/identity/memory.py:51]:
```suggestion
    @staticmethod
    def _get_identifier(target_field: str) -> Identifier:
```

i don't like closures, as they discourage unit tests

File: mex/common/identity/memory.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import hashlib
```
Comment (cutoffthetop) [inline mex/common/identity/memory.py:1]: i'd rename this to something like `HashingIdentityProvider` and trash the "database" attribute, should be way cleaner (as discussed)

**Review by cutoffthetop — CHANGES_REQUESTED** [review]

File: mex/common/identity/memory.py (hunk 1 above)
Comment (rababerladuseladim) [reply mex/common/identity/memory.py]: refactored into staticmethod in e088314

File: mex/common/identity/memory.py (hunk 2 above)
Comment (rababerladuseladim) [reply mex/common/identity/memory.py:1]: Unfortunately, this is not possible. To generate identifiers, we `hadPrimarySource` and `identifierInPrimarySource`. These parameters are optional in the current `fetch`-method implementation. Additionaly, the `fetch` method supports getting identifiers based on their stableTargetId, which is impossible with the hash-based method and no database. Hence, I will leave the database in there and will also keep the name. 
Is that alright with you?

File: mex/common/identity/memory.py (hunk 1 above)
Comment (cutoffthetop) [reply mex/common/identity/memory.py]: much obliged 🙏

File: mex/common/identity/memory.py (hunk 2 above)
Comment (cutoffthetop) [reply mex/common/identity/memory.py:1]: yep, convinced.

**Review by cutoffthetop — DISMISSED** [review]

**Review by cutoffthetop — APPROVED** [review]

**Pull request merged**